bindings/python/test/test_schema.py

```diff
@@ -70,3 +70,7 @@ def test_from_bson_units(self):
     def test_from_arrow_units(self):
         schema = Schema({"field1": int64(), "field2": timestamp("s")})
         self.assertEqual(schema.typemap, {"field1": int64(), "field2": timestamp("s")})
+
+    def test_nested_projection(self):
+        schema = Schema({"_id": int64(), "obj": {"a": int64(), "b": int64()}})
+        self.assertEqual(schema._get_projection(), {"_id": True, "obj": {"a": True, "b": True}})
```

Comment:
It would be good to add a test for the case reported in https://github.com/mongodb-labs/mongo-arrow/issues/164 too.